src/AutoBalance.cpp

```diff
@@ -503,7 +506,10 @@ class AutoBalance_AllCreatureScript : public AllCreatureScript
         if (!creature || !creature->GetMap())
             return;
 
-        if (!creature->GetMap()->IsDungeon() && !creature->GetMap()->IsBattleground() && DungeonsOnly)
+        if (DungeonsOnly && !creature->GetMap()->IsDungeon() && !creature->GetMap()->IsBattleground())
```

Comment:
I've taken the liberty to reverse this condition because if `DungeonsOnly` is set to `0` it short-cirtcuits and avoids calling `creature->GetMap()->IsDungeon()` (and potentially `creature->GetMap()->IsBattleground()` too), which should be faster to evaluate.